src/displayapp/widgets/DotIndicator.cpp

```diff
@@ -0,0 +1,25 @@
+#include "displayapp/widgets/DotIndicator.h"
+#include "displayapp/InfiniTimeTheme.h"
+
+using namespace Pinetime::Applications::Widgets;
+
+DotIndicator::DotIndicator(uint8_t nCurrentScreen, uint8_t nScreens) : nCurrentScreen {nCurrentScreen}, nScreens {nScreens} {
+}
+
+void DotIndicator::Create() {
+
+  lv_obj_t* dotIndicator[nScreens];
+  static constexpr uint8_t dotSize = 12;
+  lv_coord_t yPosition = LV_VER_RES / 2 - nScreens * 8;
+
+  for (int i = 0; i < nScreens; i++) {
+
+    dotIndicator[i] = lv_obj_create(lv_scr_act(), nullptr);
+    lv_obj_set_size(dotIndicator[i], dotSize, dotSize);
+    lv_obj_set_style_local_bg_color(dotIndicator[i], LV_BTN_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GRAY);
+    lv_obj_set_pos(dotIndicator[i], LV_HOR_RES - 16, yPosition);
+    yPosition = yPosition + 32;
```

Comment:
I'm assuming 8 should be dotSize / 2, 16 should be dotSize and 32 should be dotSize * 2.

Reply:
Kinda hard for me to explain but for the first dot its half the vertical resolution(120) minus the number of screens times eight, which is 104. so thats (224, 104) for the first dot. For the second its 32 minus the last dot which is (224, 136). They are both 16px away from the vertical center. Now if theres say three screens then its not perfectly centered but its barely noticeable. Im going to see if i can perfect it.

Reply:
Let's make LVGL do the hard work for us with containers.

```c++
void DotIndicator::Create() {
  lv_obj_t* dotIndicator[nScreens];
  static constexpr uint8_t dotSize = 12;

  lv_obj_t* container = lv_cont_create(lv_scr_act(), nullptr);
  lv_cont_set_layout(container, LV_LAYOUT_COLUMN_LEFT);
  lv_cont_set_fit(container, LV_FIT_TIGHT);
  lv_obj_set_style_local_pad_inner(container, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, dotSize);
  lv_obj_set_style_local_bg_opa(container, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_OPA_TRANSP);

  for (int i = 0; i < nScreens; i++) {
    dotIndicator[i] = lv_obj_create(container, nullptr);
    lv_obj_set_size(dotIndicator[i], dotSize, dotSize);
    lv_obj_set_style_local_bg_color(dotIndicator[i], LV_BTN_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GRAY);
  }

  lv_obj_set_style_local_bg_color(dotIndicator[nCurrentScreen], LV_BTN_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);

  lv_obj_align(container, nullptr, LV_ALIGN_IN_RIGHT_MID, 0, 0);
}

```

Reply:
Oh, that works much better, Thank you.